build/Dockerfile

```diff
@@ -2,11 +2,15 @@ FROM --platform=linux/amd64 quay.io/centos/centos:stream8 AS build
 
 ARG TARGETARCH
 
-RUN dnf install -y golang-1.17.*
+RUN mkdir /workdir
 
-COPY . /workdir
+COPY go.mod /workdir
 WORKDIR /workdir
 
+RUN dnf install -y golang-$(sed -En 's/^go +(.*+)$/\1/p' go.mod).*
+
+COPY . .
```

Comment:
What is this magic?

Reply:
We have already change to /workdir so it's not needed to be specified

Reply:
Ah of course